rules/active_ruleset.go

```diff
@@ -233,6 +233,12 @@ func (as *activeRuleSet) mappingsForNonRollupID(
 		}
 		pipelines = append(pipelines, pipeline)
 	}
+	// NB: The pipeline list should never be empty as the resulting pipelines are
+	// used to determine how the *existing* ID is aggregated and retained. If there
+	// are no rule match, the default pipeline list is used.
+	if len(pipelines) == 0 {
```

Comment:
Is there a test to catch this?

Reply:
Yeah the tests on master fails with this change, I fixed them so they are passing now.